### ISSUE TITLE
Waypoint orphans canceled deployments

### ISSUE THREAD

**Issue by izaaklauer** [hashicorp/waypoint#2636]
**Describe the bug**
If you ctrl-c in the middle of a k8s deployment, waypoint will never delete the k8s deploy object. When it comes time to prune, it will delete it from the waypoint DB, but leave the deployment around.

Ctrl-c'ing a deployment is very commonly - if a user pushes a container that crashloops, we will sit in this waiting phase for up to five minutes, and most users probably won't wait that long. We even automatically detect imagePullBackoff and warn users, which encourages them to ctrl-c.

This bug means that waypoint will pretty quickly fill clusters with deployments in CrashLoopBackoff state that are invisible in waypoint.

**Steps to Reproduce**

I observed this in k8s, but I suspect it's a problem with all plugins.

Run a waypoint deployment on k8s, and ctrl-c during the `waiting on deployment to become availabile` phase.
After the ctrl-c, you should see this:

```
$ waypoint deploy

» Deploying backend...
✓ Kubernetes client connected to https://06A15FEED4E1E0F921B6611EC8950550.gr7.us-east-2.eks.amazonaws.com with namespace dev
✓ Creating deployment...
✓ Expected "http" port "8080" for app "backend-v62"
⠸ Waiting on deployment to become available: requested=1 running=0 ready=0
! operation canceled
```

Note the deployment number (`backend-v62` in this case). Observe it in `waypoint deployment list`.

Deploy two more times. The second deploy will claim to prune the initial cancelled deployment, and that deployment will be gone from the `waypoint deployment list` view, but it will still be present in the cluster.

**Expected behavior**

When waypoint deletes the deployment from it's DB, it should also ask the plugin to delete the deployment

**Waypoint Platform Versions**
Additional version and platform information to help triage the issue if
applicable:

* Waypoint CLI Version: 0.6.1
* Waypoint Server Platform and Version: 0.6.1
* Waypoint Plugin: k8s (but likely more

**Comment by xiaolin-ninja** [comment]
Related to https://github.com/hashicorp/waypoint/issues/533 and https://github.com/hashicorp/waypoint/issues/2089

**Comment by krantzinator** [comment]
Closing in favor of #533
